bitbybit/src/bitfield/mod.rs

```diff
@@ -452,6 +452,18 @@ pub fn bitfield(args: TokenStream, input: TokenStream) -> TokenStream {
     TokenStream::from(expanded)
 }
 
+const fn mask_for_width_and_offset(width: usize, offset: usize) -> u128 {
+    if width == 128 {
+        return u128::MAX;
+    }
+    if width + offset > 128 {
+        panic!("bitfield!: Internal error: width + offset > 128");
+    }
+    // The wrapping subtraction is important for the case where width is 128.
+    // A shift with 128 would yield a value of 0, and a regular subtraction of 1 could panic.
+    ((1_u128 << width) - 1) << offset
```

Comment:
wouldn't this work and remove the need for the special case?

1u128.wrapping_shl(width).wrapping_sub(1).wrapping_shl(offset)

Reply:
Actually, your solution is probably fine. It looks more readable and makes it more obvious what's happening. So nevermind please :)